src/hardware_functions.c

```diff
@@ -117,16 +121,16 @@ void UART_PutStr(char *message)
     HAL_UART_Transmit(&UART_Handler, message, size, HAL_UART_TIMEOUT_VALUE);
 }
 
-void UART_Putc(char c)
+void UART_Putc(uint8_t c)
 {
     HAL_UART_Transmit(&UART_Handler, c, 1, HAL_UART_TIMEOUT_VALUE);
 }
 
 void UART_PutInt(uint32_t val)
 {
-    char message[UART_INT_BUFFER];
+    uint8_t message[UART_INT_BUFFER];
     uint8_t size = 0;
-    itoa(val, message, ITOA_DECIMAL);
+    sprintf((char *)message, "%ld\r\n", val);
```

Comment:
itoa is beter

Reply:
itoa heeft minder overhead en werkt sneller.
en voor de extra ondersteuning hoeven we het niet te doen